core/data/src/main/java/uk/gov/gchq/gaffer/data/elementdefinition/view/NamedViewDetail.java

```diff
@@ -336,13 +332,33 @@ public AccessPredicate getWriteAccessPredicate() {
         return writeAccessPredicate != null ? deserialise(writeAccessPredicate) : null;
     }
 
+    @JsonIgnore
+    public AccessPredicate getOrDefaultReadAccessPredicate() {
+        final AccessPredicate readAccessPredicate = getReadAccessPredicate();
+        return readAccessPredicate != null ? readAccessPredicate : getDefaultReadAccessPredicate();
```

Comment:
Here and other places you could make use of you NULL_ACCESS_PREDICATE

Reply:
Updated